Update README, clean up

denisborovikov opened the pull request:
I've updated the readme according to the latest API changes.

Please, check it. I don't get the part

```
You also have the ability to attach ref to the `TooltipTrigger` component which exposes following methods for programmatic control of the tooltip:
```

so I keep it as it is. Please, review and update it if needed.

Also, the example part should be accordingly updated.

Thank you.

mohsinulhaq commented:
`You also have the ability to attach ref to the `TooltipTrigger` component which exposes following methods for programmatic control of the tooltip:`

By this I meant that you can do the following:
```
const tooltipRef = React.createRef();

<TooltipTrigger
    ref={tooltipRef}
    tooltip={...}
>
    {...}
</TooltipTrigger>
```
Now you can use `tooltipRef.current` to show or hide the tooltip whenever you want, by calling `tooltip.current.show` and `tooltip.current.hide`.

denisborovikov commented:
What the difference between using control prop? I mean pass `tooltipShown` props from outside and control its value? Still can't get the use case of the ref.

mohsinulhaq commented:
yes, you are right, it can be controlled using the control prop (declarative). My intention was to give more (imperative) control, as you can call the methods anywhere without maintaining a state variable. For example, what if I want to close the tooltip whenever the user double clicks on a certain div. It's easier to achieve these goals using these imperative controls. Hope you got it.

mohsinulhaq commented on src/TooltipTriggerComponents.js:
```diff
@@ -1,107 +0,0 @@
-import styled from 'react-emotion';
```
I was wondering if we could keep these components as static properties of TooltipTrigger component for those who want a pre-styled arrow and box component.

denisborovikov replied:
Does it work with emotion? With styled components, it doesn't work.

I think it increases the size of the package. Do you know the size of the package with and without the emotion? According to https://bundlephobia.com/result?p=emotion@9.2.6 it's 5.7kb gzipped.

Does it worth to include the whole library just to style a single tooltip?

Anyway, It's totally up to you.

mohsinulhaq replied:
yes, i had faced a similar problem some weeks ago. innerRef was the solution. :)
I was initially using styled-components....switched to emotion only because of lower bundle size.

mohsinulhaq replied:
well no need to ship with emotion components. We can create normal static components and style them with css.

denisborovikov replied:
Let's ship styles as static CSS file as a part of the package.

denisborovikov commented:
Could you please update this part of the docs then? The docs said about `showTooltip` method but in your example, it's `tooltip.current.show`.

mohsinulhaq commented:
sorry, i wrongly wrote show and hide, it indeed is `showTooltip` and `hideTooltip`.

denisborovikov commented:
I've fixed some typos. Please, review the PR and make any updates you think needed to be done.

mohsinulhaq commented:
I have put two reviews. Have you seen them?

denisborovikov commented:
I'm trying to get it to work with styled components and it doesn't work. Works if I use regular divs in `tooltip` prop, but with styled components, popper.js rises errors.

denisborovikov commented:
Seems it won't work with styled-component, as popper uses `window.getComputedStyle()`

denisborovikov commented:
I've added static css and updated readme. I also renamed `ref` to `triggerRef` to have all ref props have consistent names: `arrorRef`, `tooltipRef`, `triggerRef`

mohsinulhaq commented:
great! now we can `npm uninstall emotion react-emotion`

dim2k2006 commented:
Hey guys! Still can not understand how programmatically close tooltip (
In my case I have "close" button right inside tooltip and I need to close the tooltip after click on this button.

<img width="439" alt="Снимок экрана 2019-03-22 в 15 00 42" src="https://user-images.githubusercontent.com/7868088/54821491-588cf300-4cb3-11e9-95d1-af0562bb9dd9.png">

mohsinulhaq commented:
You could make the tooltip controlled by creating a state variable and its onChange callback. Then you can turn that state off on `onClick` of that dismiss button.

dim2k2006 commented:
I have used "withStateHandlers" function from "recompose" library + "onVisibilityChange" callback function from "react-popper-tooltip" library. Seems to work fine. Thanks!
